cpp/fetcher/remote_peer.cc

```diff
@@ -62,10 +82,17 @@ void RemotePeer::Impl::DoneGetSTH(
     CHECK_GE(new_sth->tree_size(), 0);
   }
 
-  sth_ = new_sth;
-  if (on_new_sth_) {
-    on_new_sth_(*sth_);
+  if (!sth_ || new_sth->timestamp() > sth_->timestamp()) {
```

Comment:
Collapsed diff strikes again!

Totally agree that we're interested in same-size-but-newer, for sure, I was just worried about shrinking trees (but the `CHECK_GE` at line 80 has this covered).